src/main/java/org/tridiots/ipc/Server.java

```diff
@@ -1,60 +1,114 @@
 package org.tridiots.ipc;
 
-import java.io.EOFException;
-import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.net.ServerSocket;
-import java.net.Socket;
+import java.net.InetSocketAddress;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.Iterator;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
 public class Server implements Runnable {
-
+    private static final Logger logger = LoggerFactory.getLogger(Server.class);
     private static final int THREAD_NUMBER = 4;
-
     private ExecutorService pool = Executors.newFixedThreadPool(THREAD_NUMBER);
     private Object instance;
-    private ServerSocket serverSocket;
     private volatile boolean running = true;
 
+    private InetSocketAddress address = null;
```

Comment:
actually, Java will initialize all fields to their default values, and all `object`'s default value is `null`.
this is a little different from C++. So we don't need to assign `null` to it manually.